### PR TITLE
tests/drivers/build_all: use tc_util to unify test output

### PR DESCRIPTION
**Pull request by chase-qi**
Use the macros defined in tc_util.h to print unified test start and
result output so that test automation tool can parse test output in the
same way.

Signed-off-by: Chase Qi <chase.qi@linaro.org>

### CODE REVIEW

File: tests/drivers/build_all/src/Makefile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ccflags-y += -I${ZEPHYR_BASE}/tests/include
```
Comment (pfalcon) [inline tests/drivers/build_all/src/Makefile]: I believe there're no Makefiles in Zephyr anymore (they're generated with CMake).

Comment (chase-qi) [reply tests/drivers/build_all/src/Makefile]: Thanks for you comment! 

Without this change, './scripts/sanitycheck --platform qemu_cortex_m3 --enable-slow -b'  will hit 'tc_util.h: no such file or directory'

Comment (pfalcon) [reply tests/drivers/build_all/src/Makefile]: Good, but you would need to re-do it using CMakeLists.txt instead.

**Comment by nashif** [comment]
this particular test is not supposed to be run on anything, this is just a build test without any functionality, so it should not really report anything, the pass is achieved when all build without build errors...

**Comment by pfalcon** [comment]
> this particular test is not supposed to be run on anything

But how one (and one's CI) is supposed to know that? In our CI, we try to run everything that builds.

So, why don't we simplify it all by following a simple rule: if a test builds, it also should be able to run?

**Comment by nashif** [comment]
> But how one (and one's CI) is supposed to know that? In our CI, we try to run everything that builds.

this is being addressed, see #4724

I have a patch series to address this already, anything that is marked as build_only would be really a build_only test and should not be run.

**Comment by pfalcon** [comment]
> I have a patch series to address this already, anything that is marked as build_only would be really a build_only test and should not be run.

That addresses it only within Zephyr's CI (sanitycheck). But there're systems on top/outside of it. How they're supposed to know such facts? The easiest solution is "if it built, it can be run" and then "if it's a test which can be run, then it should have output in standard format". Any other solution is arguably more work/confusion in the long run.

**Comment by nashif** [comment]
> That addresses it only within Zephyr's CI (sanitycheck). But there're systems on top/outside of it. How they're supposed to know such facts? The easiest solution is "if it built, it can be run" and then "if it's a test which can be run, then it should have output in standard format". Any other solution is arguably more work/confusion in the long run.

You do really need to parse the testcase.yaml file and extract this information from there. If a test is labeled as build_only, then it is build_only, you can run it if you want, but that is not going to help anyone, because those tests were not designed to run and might fail. So if you report a bug, it is going to be closed. So what exactly is the purpose? 

the yaml files coming with every test can be used by any CI or system, you need to be able to parse them to make the best of the tests.

**Comment by andrewboie** [comment]
>  In our CI, we try to run everything that builds.

What? No we don't

**Comment by chase-qi** [comment]
Gents, thanks a lot for your comments and suggestions! As discussed with Paul, we will handle it in our CI code. I am closing this PR.

**Pull request closed without merging**